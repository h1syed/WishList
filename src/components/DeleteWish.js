import React, {Component} from "react";
import { deleteWish } from "../graphql/mutations";
import { API, graphqlOperation } from "aws-amplify";

class DeleteWish extends Component {

    handleDeleteWish = async wishId => {
        const input = {
            id: wishId
       }

       await API.graphql(graphqlOperation(deleteWish, {input}));
    }

    render() {
        const wish  = this.props.data;
        return(
            <button className="wish-button" onClick={() => this.handleDeleteWish(wish.id)}>Delete</button>
        )
    }
}
export default DeleteWish;