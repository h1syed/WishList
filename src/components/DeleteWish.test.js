import React, { Component } from "react";
import { shallow } from "enzyme";
import DeleteWish from "./DeleteWish";
import {API} from "aws-amplify";

describe("Delete wish component", () => {
    it("renders properly", () => {
        const wrapper = shallow(<DeleteWish/>);
        expect(wrapper).toMatchSnapshot();
    });

    describe("Delete button" , () => {
        it("should call handleDeleteWish on click", async () => {
            const mockWish = {
                id: "123"
            }
            const wrapper = shallow(<DeleteWish data = {mockWish}/>);
            const spy = jest.spyOn(wrapper.instance(), "handleDeleteWish");
            const spy1 = jest.spyOn(API, "graphql").mockReturnValue(Promise.resolve());            
            wrapper.find({ className: "wish-button"}).simulate("click");
            expect(spy).toHaveBeenCalledTimes(1);
            expect(spy).toHaveBeenCalledWith(mockWish.id);
            spy.mockRestore();
        });
    });
});