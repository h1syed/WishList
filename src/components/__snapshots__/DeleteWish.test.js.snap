// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete wish component renders properly 1`] = `ShallowWrapper {}`;
